components/script/dom/htmlscriptelement.rs

```diff
@@ -274,7 +274,8 @@ impl HTMLScriptElement {
 
         let window = window_from_node(self);
         let window = window.r();
-        let base_url = window.get_url();
+        let document = document_from_node(self);
+        let base_url = document.base_url();
```

Comment:
It actually is used (some lines above). There is a variable called `doc`. I'll work on your (and @Ms2ger) comments later (after work).